### ISSUE TITLE
Difference between bit8 and uint8

### ISSUE THREAD

**Issue by nick-hunter** [keichi/binary-parser#239]
I'm currently writing an application that creates several hundred different parsers based on structures defined in a database. I have a lookup table that converts the datatype from the DB into a type from `binary-parser`. I discovered a difference between `bit8` and `uint8` and I'm wondering if this is expected behavior or not.

```
const buff = Buffer.from("7B2D", "hex");
const p1 = new Parser().bit1("one").uint8("eight") // { one: 0, eight: 45 }
const p2 = new Parser().bit1("one").bit8("eight") // { one: 0, eight: 246 }
```

Here is the generated code for both parsers
```
// p1
var dataView = new DataView(buffer.buffer, buffer.byteOffset, buffer.length);
var offset = 0;
var vars = {};
vars.$parent = null;
vars.$root = vars;
var $tmp0 = 0;
$tmp0 = dataView.getUint8(offset);
offset += 1;
vars.one = $tmp0 >> 7 & 0x1;
vars.eight = dataView.getUint8(offset, false);
offset += 1;
delete vars.$parent;
delete vars.$root;
return vars;

// p2
var dataView = new DataView(buffer.buffer, buffer.byteOffset, buffer.length);
var offset = 0;
var vars = {};
vars.$parent = null;
vars.$root = vars;
var $tmp0 = 0;
$tmp0 = dataView.getUint16(offset);
offset += 2;
vars.one = $tmp0 >> 15 & 0x1;
vars.eight = $tmp0 >> 7 & 0xff;
delete vars.$parent;
delete vars.$root;
return vars;
```

The code for p1 reads one bit, skips 7 bits, and starts reading at the start of the next byte. Is this expected? I didn't see anything in the docs about this difference. I'm thinking about tracking both the bit and byte position, and rewriting uint8 to bit8 if the bit offset != 0. Just wanted to see if anyone else has run into similar issues.

**Comment by keichi** [comment]
Yes, this is by design and we should document it. Basically, anything except bit fields needs to be byte-aligned. This is to simplify the parser builder code (otherwise we would need to keep track of bit and byte positions as you mentioned).

**Comment by keichi** [comment]
This is expected behavior. `uint8` always reads a full byte-aligned value, while `bit8` reads 8 bits from the current position in a continuous bit stream.

When you use `bit1` followed by `uint8`, the remaining 7 bits of the first byte are discarded, and `uint8` reads the next full byte. With `bit1` followed by `bit8`, the 8 bits are read immediately after the first bit, spanning across byte boundaries.

In your example:
- `bit1("one").uint8("eight")` on `0x7B2D`: reads bit 7 of `0x7B` → 0, then reads full byte `0x2D` → 45
- `bit1("one").bit8("eight")` on `0x7B2D`: reads bit 7 of `0x7B2D` → 0, then reads next 8 bits → 246 (`0xF6`)

Closing as expected behavior.
